File: docs/_code/kafka_consumer.cpp
#include <mofka/KafkaDriver.hpp>
#include <mofka/TopicHandle.hpp>
#include <iostream>

int main(int argc, char** argv) {

    if(argc != 3) {
        std::cerr << "Usage: "
                  << argv[0] << " <config-file> <topic>" << std::endl;
        return -1;
    }

    auto config_file = argv[1];
    auto topic_name  = argv[2];

    try {

        mofka::KafkaDriver driver{config_file};

        mofka::TopicHandle topic = driver.openTopic(topic_name);

        mofka::Consumer consumer = topic.consumer("consumer");
        for(size_t i = 0; i < 100; ++i) {
            mofka::Event event = consumer.pull().wait();
            std::cout << event.id() << ": " << event.metadata().string() << std::endl;
            if((i+1) % 10 == 0) event.acknowledge();
        }

    } catch(const mofka::Exception& ex) {
        std::cerr << ex.what() << std::endl;
    }

    return 0;
}

File: docs/_code/kafka_consumer.py
import sys
import json
import pymargo.core
from pymargo.core import Engine
from mochi.mofka.client import KafkaDriver


def consume(config_file: str, topic_name: str):
    driver = KafkaDriver(config_file)
    topic = service.open_topic(topic_name)
    consumer = topic.consumer(name="myconsumer")

    for i in range(0, 100):
        event = consumer.pull().wait()
        print(event.metadata)
        if (i+1) % 10:
            event.acknowledge()


if __name__ == "__main__":

    if len(sys.argv) != 3:
        print(f"Usage: {sys.argv[0]} <config-file> <topic>")
    config_file = sys.argv[1]
    topic_name = sys.argv[2]

    consume(config_file, topic_name)
